the-collab-lab/tcl-73-smart-shopping-list
# Add Delete List Functionality

## Description

Currently, users can create lists on the home page, but there is no option to remove them if they are no longer needed. This PR aims to enhance the user experience by adding a delete option for lists on the home page.
When users click the delete button, they'll receive a confirmation message reminding them that deleting the list will also delete all items within it. This extra step helps prevent accidental deletions and gives users a chance to reconsider. With this feature, users will have more control over managing their lists and the opportunity to delete a list if they no longer need it.

## Related Issue
Closes #17

## Acceptance Criteria

- [ ] On the home page, add a delete button to every list the user has on the home page.
- [ ] If the user clicks delete, the system should show a message asking if the user is sure they want to delete the list and inform them that all items in the selected list will also be deleted. If the user confirm their choice, delete the list along with all its items.
- [ ] Create a deleteList function in firebase.js to remove a list from the Firestore database.

## Type of Changes

enhancement

## Updates

### Before

<img width="384" alt="Screenshot 2024-04-01 at 1 54 48 PM" src="https://github.com/the-collab-lab/tcl-73-smart-shopping-list/assets/88466266/257e983e-32b7-4bc4-8c73-2d95da7a1aad">

### After

![delete list btn](https://github.com/the-collab-lab/tcl-73-smart-shopping-list/assets/88466266/11cdc1e0-60b4-4028-a398-59fe960de80d)
![delete list2](https://github.com/the-collab-lab/tcl-73-smart-shopping-list/assets/88466266/8bf8c1e9-aa35-4dea-8069-d9c96b467b68)
![delete 3](https://github.com/the-collab-lab/tcl-73-smart-shopping-list/assets/88466266/5a7b1b9f-0b18-4c62-9808-a99786cc8c9d)
![list 4](https://github.com/the-collab-lab/tcl-73-smart-shopping-list/assets/88466266/007ee2b6-2063-4719-8ce4-32cb595d7522)
Note:
User can delete only their own list:
![delete list 5](https://github.com/the-collab-lab/tcl-73-smart-shopping-list/assets/88466266/de0f9b9f-c52b-4685-8c36-f569ba9e359d)

## Testing Steps / QA Criteria

git pull
git checkout it-delete-single-list
npm ci
npm start

- Navigate to the Home page.
- Select a list and click on the delete button next to it.
- Check if you see a message asking the user if they want to delete the selected list.
- Select 'cancel' and check if the list is still present.
- Repeat the previous steps and select 'yes'. Check if the list was deleted and if you see an alert on your screen with a message stating that the selected list was successfully deleted.
- Try to delete a shared list with you. I have disabled deleting a list for people who have access to it but are not the owners
- Navigate to Firebase and check if the list was deleted. If you see a list in the database, click on the name. You should see a message: "This document does not exist; it will not appear in queries or snapshots."
